pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/AbstractSerdeIndexContract.java

```diff
@@ -103,6 +106,14 @@ protected <C extends IndexConfig> C getActualConfig(String column, IndexType<C,
     return confMap.get(column).getConfig(type);
   }
 
+  protected <C extends IndexConfig> C getActualFromIndexLoadingConfig(String column, IndexType<C, ?, ?> type) {
+    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig(_tableConfig, _schema);
+    Map<String, FieldIndexConfigs> confMap =
+        FieldIndexConfigsUtil.createIndexConfigsByColName(_tableConfig, _schema, indexLoadingConfig::getDeserializer);
+
+    return confMap.get(column).getConfig(type);
+  }
```

Comment:
This can be changed with:
```java
  protected <C extends IndexConfig> C getActualFromIndexLoadingConfig(String column, IndexType<C, ?, ?> type) {
    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig(_tableConfig, _schema);
    return indexLoadingConfig.calculateIndexConfigsByColName()
        .get(column)
        .getConfig(type);
  }
```

By doing that we don't need to make `IndexingLoadingConfig.getDeserializer` public, so the encapsulation is better.